### PR TITLE
HSEARCH-2690 FileHelper based index replication attempts to delete lo…

### PR DESCRIPTION
**Pull request by Sanne**
…cked files on Windows

I decided to change the semantics of the method, even though we occasionally use it for other purposes as those are just testing. Adding a new method for this seemed overkill.

 - https://hibernate.atlassian.net/browse/HSEARCH-2690

### CODE REVIEW

**Comment by Sanne** [comment]
@Hibernate-CI re test this please

**Comment by yrodiere** [comment]
Jenkins, retest this please

File: engine/src/main/java/org/hibernate/search/util/logging/impl/Log.java [hunk 1]
```diff
@@ -1029,4 +1029,8 @@
 	@Message(id = 335, value = "Unable to build a Lucene query from the query string '%1$s'.")
 	EmptyQueryException unableToBuildLuceneQueryFromQueryString(String query);
 
+	@LogMessage(level = Level.DEBUG)
+	@Message(id = 336, value = "A file could not be deleted: likely lock contention. Not a problem for index replications as it will be attempted again in the future.")
+	void fileDeleFailureIgnored(@Cause IOException e);
```
Comment (yrodiere) [inline engine/src/main/java/org/hibernate/search/util/logging/impl/Log.java]: Shouldn't this be "fileDeleteFailureIgnored"? I don't get why we would use an abbreviation here.

File: engine/src/main/java/org/hibernate/search/util/impl/FileHelper.java [hunk 2]
```diff
@@ -184,10 +184,11 @@ public static void delete(File file) throws IOException {
...
 	/**
 	 * Attempts to delete a file. If the file is a directory delete recursively all content.
+	 * Any IOException preventing a file to be deleted will be swallowed.
 	 *
 	 * @param path the file or directory to be deleted
 	 *
-	 * @throws IOException if it wasn't possible to delete all content which is a common problem on Windows systems.
+	 * @throws IOException on unexpected io errors
 	 */
 	public static void delete(Path path) throws IOException {
```
Comment (yrodiere) [inline engine/src/main/java/org/hibernate/search/util/impl/FileHelper.java]: Given the  new behavior, maybe renaming this to "tryDelete" would make sense?

File: engine/src/main/java/org/hibernate/search/util/logging/impl/Log.java (hunk 1 above)
Comment (Sanne) [inline engine/src/main/java/org/hibernate/search/util/logging/impl/Log.java]: thanks yes that's just a typo

File: engine/src/main/java/org/hibernate/search/util/impl/FileHelper.java (hunk 2 above)
Comment (Sanne) [inline engine/src/main/java/org/hibernate/search/util/impl/FileHelper.java]: ok; In that case I would deprecate it.

**Comment by Sanne** [comment]
thanks, good ideas. Applied them, then actually differentiated the methods to make sure there's no mistake and merged.

**Pull request merged**